pkg/aws_config_client/completer.go

```diff
@@ -114,49 +114,126 @@ func (c *completer) SurveyProfile() (*AWSNamedProfile, error) {
 		AWSProfile: profile,
 	}
 
-	return namedProfile, nil
+	return []*AWSNamedProfile{namedProfile}, nil
+}
+
+// SurveyRole will ask a user to configure a default role
+func (c *completer) SurveyRoles() ([]*AWSNamedProfile, error) {
+	// first prompt for roles
+	roles := c.awsConfig.GetRoleNames()
+	accounts := c.awsConfig.GetAccounts()
+
+	roleIdx, err := c.prompt.Select(
+		"Select the AWS role you would like to make default:",
+		roles,
+	)
+	if err != nil {
+		return nil, err
+	}
+	targetRole := roles[roleIdx]
+
+	configuredProfiles := []*AWSNamedProfile{}
+
+	for _, account := range accounts {
+		profileName := c.calculateDefaultProfileName(account)
+
+		// get the roles associated with this account
+		profiles := c.awsConfig.GetProfilesForAccount(account)
+		for _, profile := range profiles {
+
+			// Initialize a new AWSNamedProfile
+			currentProfile := AWSNamedProfile{
+				AWSProfile: server.AWSProfile{
+					ClientID:   profile.ClientID,
+					AWSAccount: profile.AWSAccount,
+					RoleARN:    profile.RoleARN,
+					IssuerURL:  profile.IssuerURL,
+				},
+			}
+
+			currentProfile.Name = fmt.Sprintf("%s-%s", profileName, profile.RoleName)
+			configuredProfiles = append(configuredProfiles, &currentProfile)
+
+			if profile.RoleName == targetRole {
+				defaultProfile := AWSNamedProfile{
+					Name: profileName,
+					AWSProfile: server.AWSProfile{
+						ClientID:   profile.ClientID,
+						AWSAccount: profile.AWSAccount,
+						RoleARN:    profile.RoleARN,
+						IssuerURL:  profile.IssuerURL,
+					},
+				}
+				configuredProfiles = append(configuredProfiles, &defaultProfile)
+			}
+		}
+	}
+	return configuredProfiles, nil
+}
+
+func (c *completer) Survey() ([]*AWSNamedProfile, error) {
+	configureOptions := []string{"Role", "Profile"}
+	configureIdx, err := c.prompt.Select("Would you like to configure AWS by role or profile?", configureOptions)
```

Comment:
How do you feel about the following wording?
```suggestion
	configureOptions := []string{
	   "Automatically configure the same role for each account?", 
	   "Configure one role at a time?"}
	 configureFuncs := []func()...{c.SurveyProfile, c.SurveyRole}
	configureIdx, err := c.prompt.Select("How would you like to configure your AWS config?", configureOptions)
	
	return configureFuncs[configureIdx]()
```